htcanalyze/resource.py

```diff
@@ -0,0 +1,133 @@
+            f"Usage: [{self.get_color()}]{self.usage}[/{self.get_color()}], " \
+            f"Requested: {self.requested}, " \
+            f"Allocated: {self.allocated}"
+        return s_res
+
+    def chg_lvl_on_threholds(self, bad_usage, tolerated_usage):
+        """Set warning level depending on thresholds."""
+        if self.requested != 0:
+            deviation = self.usage / self.requested
+
+            if str(self.usage) == 'nan':
+                self.warning_level = 'light_warning'
+            elif not 1 - bad_usage <= deviation <= 1 + bad_usage:
+                self.warning_level = 'error'
+            elif not 1 - tolerated_usage <= deviation <= 1 + tolerated_usage:
+                self.warning_level = 'warning'
+            else:
+                self.warning_level = 'normal'
+        elif self.usage > 0:  # Usage without any request ? NOT GOOD
+            self.warning_level = 'error'
+        else:
+            self.warning_level = 'normal'
+
+
+def create_avg_on_resources(
+        job_resource_list: List[List[Resource]]
+) -> List[Resource]:
+    """
+    Create a new List of Resources in Average.
+
+    :param job_resource_list: list(list(Resource))
+    :return: list(Resource)
+    """
+    n_jobs = len(job_resource_list)
+    res_cache = dict()
+
+    # calc total
+    for job_resources in job_resource_list:
+        for resource in job_resources:
+            # add resource
+            if res_cache.get(resource.name):
+                res_cache[resource.name].usage += ntn(resource.usage)
+                res_cache[resource.name].requested += ntn(resource.requested)
+                res_cache[resource.name].allocated += ntn(resource.allocated)
+            # create first entry
+            else:
+                res_cache[resource.name] = Resource(resource.name,
+                                                    ntn(resource.usage),
+                                                    ntn(resource.requested),
+                                                    ntn(resource.allocated))
+
+    avg_res_list = list(res_cache.values())
+
+    # calc avg
+    for resource in avg_res_list:
+        resource.name = "Average " + resource.name
+        resource.usage = round(resource.usage / n_jobs, 3)
+        resource.requested = round(resource.requested / n_jobs, 2)
+        resource.allocated = round(resource.allocated / n_jobs, 2)
+
+    return avg_res_list
+
+
+def refactor_resources(resources: dict) -> List[Resource]:
+    """Convert a dict of lists to a list of Resource objects."""
+    resources = {k.lower(): v for k, v in resources.items()}
+    resources["name"] = resources.pop("resources")
+    resources = [dict(zip(resources, v)) for v in zip(*resources.values())]
+    resources = [Resource(**resource) for resource in resources]
+    return resources
+
+
+def convert_res_to_dict(resources: List[Resource]) -> dict:
+    """Convert a list of Resource back to this dict scheme."""
+    res_dict = {
```

Comment:
can just do `return { ... }`, rather than assign it to `res_dict` and then return `res_dict`.